pkgs/build-support/bintools-wrapper/ld-wrapper.sh

```diff
@@ -232,8 +232,11 @@ fi
 
 # Only add --build-id if this is a final link. FIXME: should build gcc
 # with --enable-linker-build-id instead?
+#
+# Note: `lld` interprets `--build-id` to mean `--build-id=fast`; GNU ld defaults
+# to SHA1.
 if [ "$NIX_SET_BUILD_ID_@suffixSalt@" = 1 ] && ! (( "$relocatable" )); then
-    extraAfter+=(--build-id)
+    extraAfter+=(--build-id="${NIX_BUILD_ID_STYLE:-sha1}")
```

Comment:
Is there some ld-wrapper doc we should update with NIX_BUILD_ID_STYLE?

Reply:
Oh good point!

Maybe the [`separateDebugInfo` section in the manual](https://nixos.org/manual/nixpkgs/stable/#stdenv-separateDebugInfo) is a good place for this? It actually _already_ talks about how build-id is a SHA1 hash :slightly_smiling_face:

Reply:
Please write some documentation in a follow-up PR!